src/commands/data/schemas/util/schemaverify.ts

```diff
@@ -189,3 +193,28 @@ export class SchemaVerify {
     return { ok, errors };
   }
 }
+
+// The data service automatically assigns an id to all objects in an array
+// This function checks if the schema has an array of objects with an id property conflicting the one from the data service
+function getIdInObjectArrayErrors(properties: any, path = []) {
+  const pathsWithIdInArray = [];
+  for (const [name, value] of Object.entries<any>(properties)) {
+    // For all properties with type array of objects, check if they have an id property
+    if (value.type === 'array' && value.items.type === 'object') {
```

Comment:
is it correct to state that arrays of arrays of objects are not covered? Arrays of objects are recursively traversed, but arrays of arrays are not I belief 🤔

I know about a use for FibriCheck in their measurement schema, a 2d array representing "pixels" in a "video" matrix:
```js
quadrants: [
  [
    { 
      id: 'should not be customizable',
      y: [1,2,3,4,5],
      u: [1,2,3,4,5],
      v: [1,2,3,4,5]
    },
    {...},
    {...},
    {...}
  ],
  [
    {...},
    {...},
    {...},
    {...}
  ],
  [...],
  [...],
]
```

Reply:
@wwaaijer-exh Here is my logic (BIG WIP) for the validation of properties.
https://github.com/ExtraHorizon/exh-cli/blob/feature/SVCS-2143-The-schema-verification-is-too-strict-in-validating-condition-properties/src/commands/data/schemas/util/schemaverify.ts#L165

This ticket actually (to me) feels like it could be a one liner fit into here 🤔 Maybe we need to re-run the same function when validation properties too?

Reply:
Yes, very similar! Although with a different purpose 🤔 I would not go to too much trouble to combine these two just yet. The logic here is still pretty simple, even with a fix for the problem I pointed out. In my opinion some duplication might not be that bad if it serves simplicity. But happy to discuss or change my mind!

Reply:
I would wait with merging them until at least both tickets are done 😄  Then we can easily review both validation features separately without the code becoming too big. If we want to merge afterwards, then it should be a simple re-factoring PR.